metaflow/plugins/kfp/kfp.py

```diff
@@ -369,65 +369,65 @@ def create_run_on_argo(
         """
         Creates a new run on Argo using the `KubernetesClient()`.
         """
-        workflow_template: Dict[str, Any] = self._create_workflow_yaml(
-            flow_parameters, output_format="argo-workflow"
+        workflow: Dict[str, Any] = self._create_workflow_yaml(
+            flow_parameters, kind="Workflow", max_run_concurrency=max_run_concurrency
         )
+        argo_workflow_name: str = sanitize_k8s_name(self.name)
 
-        try:
-            argo_workflow_name = sanitize_k8s_name(self.name)
+        config_map: Dict[str, Any] = KubeflowPipelines._config_map(
+            argo_workflow_name, max_run_concurrency
+        )
 
-            config_map: Dict[str, Any] = KubeflowPipelines._create_config_map(
-                argo_workflow_name, max_run_concurrency
-            )
+        try:
             # Create the Argo synchronization ConfigMap
             config = ArgoClient(
                 namespace=kubernetes_namespace
             ).create_workflow_config_map(argo_workflow_name, config_map)
 
             # Create/Run the Argo Workflow
-            workflow = ArgoClient(namespace=kubernetes_namespace).run_workflow(
-                workflow_template
+            running_workflow = ArgoClient(namespace=kubernetes_namespace).run_workflow(
+                workflow
             )
-            return workflow, config
+            return running_workflow, config
         except Exception as e:
             raise KfpException(str(e))
 
-    def create_workflow_yaml_file(
+    def write_workflow_kind(
         self,
         output_path: str,
         flow_parameters: Optional[dict] = None,
-        output_format: str = "argo-workflow",
+        kind: str = None,
```

Comment:
It looks like `kind` should be a required argument. Shall we remove the default `None` value?